adafruit_wiznet5k/adafruit_wiznet5k_socket.py

```diff
@@ -181,6 +181,11 @@ def bind(self, address):
             if ip_address != current_ip:
                 _the_interface.ifconfig = (ip_address, subnet_mask, gw_addr, dns)
         self._listen_port = address[1]
+        # For UDP servers we need to open the socket here because we won't call 
+        # listen
+        if self._sock_type == SOCK_DGRAM:
```

Comment:
In CPython all I have to call is:
```
sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
sock.bind((HOST, PORT))
```
However, the Wiznet5k needs to be instructed to open a socket on the chip. For TCP that happens during the `socket.listen()` call. No such thing for UDP, so I've decided to open the socket on the chip during `bind()`.

Reply:
This is OK by me since this library should mimic CPython `socket` and calling `bind` on a UDP socket is possible for a CPython UDP socket [1].

[1] https://wiki.python.org/moin/UdpCommunication